internal/resolve/catalog.go

```diff
@@ -190,6 +178,42 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 	return resolvedBundle, resolvedBundleVersion, priorDeprecation, nil
 }
 
+type resolutionErrorInfo struct {
+	PackageName     string
+	Version         string
+	Channels        []string
+	InstalledBundle *ocv1alpha1.BundleMetadata
+	ResolvedBundles []foundBundle
+}
+
+func buildResolutionError(rei resolutionErrorInfo) error {
+	var errStringBuilder strings.Builder
+	if rei.InstalledBundle != nil {
+		errStringBuilder.WriteString(fmt.Sprintf("error upgrading from currently installed version %q: ", rei.InstalledBundle.Version))
+	}
+
+	if len(rei.ResolvedBundles) > 1 {
+		matchedCatalogs := []string{}
+		for _, r := range rei.ResolvedBundles {
+			matchedCatalogs = append(matchedCatalogs, r.catalog)
+		}
+		slices.Sort(matchedCatalogs) // sort for consistent error message
+		errStringBuilder.WriteString(fmt.Sprintf("package %q found in multiple catalogs: %v", rei.PackageName, matchedCatalogs))
+	} else {
+		errStringBuilder.WriteString(fmt.Sprintf("no package %q found ", rei.PackageName))
```

Comment:
Looking at the tests, I wonder if we could improve terminology a bit in these cases. I feel like the current error messages have some ambiguity in the bundle vs package terms.

Change this:
```
package "foo" found in multiple catalogs: [fizz buzz] matching version ">=1.0.0"
```
To:
```
found bundles in package "foo" matching version ">=1.0.0" in channels [a b] in multiple catalogs of the same priority: [fizz buzz]
```


Change this:
```
no package "foo" found matching version ">=1.0.0" in channels [a b]
```
To:
```
no bundles found in package "foo" matching version ">=1.0.0" in channels [a b]
```